spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/GatewayAutoConfiguration.java

```diff
@@ -483,8 +484,8 @@ public RewriteResponseHeaderGatewayFilterFactory rewriteResponseHeaderGatewayFil
 	}
 
 	@Bean
-	public SetStatusGatewayFilterFactory setStatusGatewayFilterFactory() {
-		return new SetStatusGatewayFilterFactory();
+	public SetStatusGatewayFilterFactory setStatusGatewayFilterFactory(@Value("${spring.cloud.gateway.original-status.header.name:original-http-status}") String originalStatusHeaderName) {
```

Comment:
I would prefer we use configuration properties

Reply:
I don't prefer 😀, it should use them.